src/Amazon.AspNetCore.DataProtection.SSM/PersistOptions.cs

```diff
@@ -28,5 +27,12 @@ public class PersistOptions
         /// don't specify a key ID, the system uses the default key associated with your AWS account.
         /// </summary>
         public string KMSKeyId { get; set; }
+
+        /// <summary>
+        /// Highest storage tier that can get used. Higher tiers allow more characters which is required for larger keys.
+        /// Even if a higher tier is configured, the lowest possible tier will be chosen when creating the parameter.
+        /// The default tier is 'Standard' (4096 characters).
+        /// </summary>
+        public ParameterTier TierStorageMode { get; set; } = ParameterTier.Standard;
```

Comment:
I would like to create a new enum instead of reusing `ParameterTier` from the SDK. The most important reason is if a new tier gets added unless this library is updated setting new values will have an undefined behavior.

I would like an enum with the following values:
```csharp
publiic enum StorageMode
{
   StandardOnly, // The default and will only use standard tier of fail if the value won't fit.
   AdvancedUpgradeable, // Use advanced tier if the value won't fit in standard tier
   AdvancedOnly // Only use advanced tier. 
}
```

The `AdvancedOnly` mode is not covered yet in this PR. I can imagine some users having a need to only ever use Advanced mode. For example they have used up all of their free standard parameters or they want to take advantage of parameter policies. If you don't want to implement `AdvancedOnly` now that is okay I can take care of that later. But by having this separate enum we have the possibility to add it later.

Reply:
Implementing that should be no problem.

Is the naming consistend when the parameter name is TierStorageMode but the enum name is only StorageMode? I would suggest naming both StorageTierMode -> It is the Mode of the Storage Tier that is beeing used.

Reply:
Agreed on the name of using `TierStorageMode`